scipy/stats/_binom_test.py

```diff
@@ -0,0 +1,165 @@
+from collections import namedtuple
+import numpy as np
+from scipy.optimize import brentq
+from ._discrete_distns import binom
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+
+class BinomTestResult:
+
+    def __init__(self, pvalue, proportion_estimate, proportion_ci):
+        self.pvalue = pvalue
+        self.proportion_estimate = proportion_estimate
+        self.proportion_ci = proportion_ci
+
+    def __repr__(self):
+        s = (f"BinomTestResult(pvalue={self.pvalue}, "
+             f"proportion_estimate={self.proportion_estimate}, "
+             f"proportion_ci={self.proportion_ci})")
+        return s
+
+
+def _binom_conf_int(x, n, confidence_level, alternative):
+    """
+    Compute the estimate and confidence interval for the binomial test.
+
+    Returns p, p_low, p_high
+    """
+    if alternative == 'two-sided':
+        alpha = (1 - confidence_level) / 2
+        plow = brentq(lambda p: binom.sf(x-1, n, p) - alpha, 0, 1)
+        phigh = brentq(lambda p: binom.cdf(x, n, p) - alpha, 0, 1)
+    elif alternative == 'less':
+        alpha = 1 - confidence_level
+        plow = 0.0
+        phigh = brentq(lambda p: binom.cdf(x, n, p) - alpha, 0, 1)
+    elif alternative == 'greater':
+        alpha = 1 - confidence_level
+        plow = brentq(lambda p: binom.sf(x-1, n, p) - alpha, 0, 1)
+        phigh = 1.0
+    estimate = x/n
+    return estimate, plow, phigh
+
+
+def binom_test(x, n=None, p=0.5, alternative='two-sided',
+               *, confidence_level=None):
+    """
+    Perform a test that the probability of success is p.
```

Comment:
Documentation looks good. I guess we can't change the names of x and p now.

Reply:
I changed `x` to `k`.  I'm on the fence about changing `p` to `pi`.